src/unix/core.c

```diff
@@ -1102,3 +1102,66 @@ int uv_os_homedir(char* buffer, size_t* size) {
 
   return 0;
 }
+
+
+int uv_os_tmpdir(char* buffer, size_t* size) {
+  char* buf;
+  size_t len;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  #define CHECK_ENV_VAR(name) \
```

Comment:
Style issues: don't indent the `#define` directive (technically disallowed by the spec) and wrap the body in a `do { } while (0)` (without a trailing semicolon.)